packages/components/src/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,181 @@
+  initialOptions = [],
+  value,
+  onChange: onSelectOption,
+  getOptions,
+  placeholder,
+}: AutocompleteProps) {
+  const [options, setOptions] = useState(initialOptions);
+  const [optionsMenuVisible, setOptionsMenuVisible] = useState(false);
+  const [text, setText] = useState((value && value.label) || "");
+
+  const inputRef = useRef() as MutableRefObject<InputTextRef>;
+
+  useEffect(() => {
+    if (value) {
+      setText(value.label);
+    } else {
+      setText("");
+      setOptions(initialOptions);
+    }
+  }, [value]);
+
+  return (
+    <div className={styles.autocomplete}>
+      <InputText
+        value={text}
+        onChange={textChange}
+        placeholder={placeholder}
+        onFocus={handleTextFocus}
+        onBlur={handleTextBlur}
+        ref={inputRef}
+      />
+      <Menu
+        visible={optionsMenuVisible}
+        options={options}
+        activeOption={value}
+        onOptionSelect={selectOption}
+      />
+    </div>
+  );
+
+  function selectOption(chosenOption: Option) {
+    onSelectOption(chosenOption);
+    textChange(chosenOption.label);
+    setOptionsMenuVisible(false);
+  }
+
+  async function textChange(newValue: string) {
+    setOptionsMenuVisible(true);
+    setText(newValue);
+    const results = await getOptions(newValue);
+    setOptions(results);
+  }
+
+  function handleTextBlur() {
+    setOptionsMenuVisible(false);
+    if (value == undefined || value.label !== text) {
+      onSelectOption(undefined);
+    }
+  }
+
+  function handleTextFocus() {
+    setOptionsMenuVisible(true);
+  }
+}
+
+interface MenuProps {
+  visible: boolean;
+  options: Option[];
+  activeOption?: Option;
+  onOptionSelect(chosenOption: Option): void;
+}
+
+function Menu({ visible, options, activeOption, onOptionSelect }: MenuProps) {
+  const [selectedIndex, setSelectedIndex] = useState(-1);
+  const optionMenuClass = classnames(styles.options, {
+    [styles.visible]: visible,
+  });
+
+  setupKeyListeners();
+
+  useEffect(() => setSelectedIndex(-1), [options]);
+
+  return (
+    <div className={optionMenuClass}>
+      {options.map((option, index) => {
+        const optionClass = classnames(styles.option, {
+          [styles.active]: index === selectedIndex,
+        });
+        return (
+          <button
+            className={optionClass}
+            key={option.value}
+            onMouseDown={handleMouseDown(option)}
+          >
+            <Text>
+              {activeOption && option.value === activeOption.value && (
+                <>
+                  <Icon name="checkmark" size="small" />{" "}
+                </>
+              )}
+              {option.label}
+            </Text>
+          </button>
+        );
+      })}
+    </div>
+  );
+
+  function handleMouseDown(option: Option) {
+    return () => {
+      onOptionSelect(option);
+    };
+  }
+
+  function setupKeyListeners() {
+    useEffect(() => {
+      const handleKeyDown = (event: { key: string }) => {
```

Comment:
Destructuring and types make for confusing function definitions now. Is this declaring that `event` is of type, `{ key: string }` or is it doing destructuring? I assume it has to be a type definition because you're using `event` later and not `key`.

Reply:
This anonymous interface was a remnant of the old component. I have updated it to reflect `KeyboardEvent` as it should.